manifests/network.pp

```diff
@@ -0,0 +1,38 @@
+# -- Define: systemd::network
+# Creates network config for systemd-networkd
+define systemd::network (
+  Enum['file', 'absent'] $ensure         = file,
+  Stdlib::Absolutepath $path             = '/etc/systemd/network',
+  Optional[String] $content              = undef,
+  Optional[String] $source               = undef,
+  Optional[Stdlib::Absolutepath] $target = undef,
+  Boolean $restart_service               = true,
+  Boolean $manage_service                = true,
+){
+
+  include ::systemd
+
+  if $manage_service {
+    service{'systemd-networkd':
```

Comment:
I was thinking of a ```systemd::networkd``` class that's included here, but since configuration goes through init that could work as well.